### ISSUE TITLE
AsyncPipe template compilation error if value is Promise|Observable

### ISSUE THREAD

**Issue by MikeMatusz** [angular/angular#27369]
# 🐞 bug report

### Affected Package
The issue is caused by package @angular/core

### Is this a regression?
No.

### Description
Similar to #22100, but if a value is typed as (Promise<any> | Observable<any>) the template compile fails.  I also like the [comment](https://github.com/angular/angular/issues/22100#issuecomment-364294572) on that issue about accepting non-async values as pass-through similar to the async keyword, though that would technically be a feature request.  That's actually the exact functionality I was trying to implement by using an intermediate pipe when I ran into this issue.

As a workaround, I'm converting Promises to Observables inside my Pipe so the type isn't ambiguous, but it seems like that shouldn't be necessary since AsyncPipe.transform has a signature that should accept either.

## 🔬 Minimal Reproduction
https://stackblitz.com/edit/angular-issue-repro2-kpsdeu

## 🔥 Exception or Error
<pre><code>
Argument of type 'Promise<any> | Observable<any>' is not assignable to parameter of type 'Promise<any>'.
Type 'Observable<any>' is not assignable to type 'Promise<any>'.
Property 'then' is missing in type 'Observable<any>'.
</code></pre>

## 🌍  Your Environment

**Angular Version:**
<pre><code>
Node: 8.11.3
OS: win32 x64
Angular: 6.1.10
... animations, common, compiler, compiler-cli, core, forms
... http, language-service, platform-browser
... platform-browser-dynamic, platform-server, router, upgrade

@angular/cdk: 6.1.0
@angular/material: 6.1.0
@angular-devkit/build-optimizer: 0.0.42
@angular-devkit/core: 0.0.29
@angular-devkit/schematics: 0.0.52
@ngtools/json-schema: 1.1.0
@ngtools/webpack: 1.9.6
@schematics/angular: 0.1.17
typescript: 2.9.2
webpack-bundle-analyzer: 2.9.0
webpack-karma-die-hard: 1.0.4
webpack-merge: 4.1.0
webpack: 3.10.0
</code></pre>

**Comment by dawidgarus** [comment]
From what I see async pipe can accept either Promise or Observable, but not Promise|Observable.
I think this two lines:
https://github.com/angular/angular/blob/7.1.1/packages/common/src/pipes/async_pipe.ts#L89-L90
should be merged into:
```typescript
transform<T>(obj: Observable<T>|Promise<T>|null|undefined): T|null;
```

**Comment by petebacondarwin** [comment]
Should be fixed by #36277

**Comment by jessicajaniuk** [comment]
Thanks for reporting this issue. This issue is now obsolete due to changes in the recent releases. Please update to the most recent Angular version.

If the problem still exists in your application, please [open a new issue](https://github.com/angular/angular/issues/new/choose) and follow the instructions in the issue template.
